crates/net/src/eventsource/mod.rs

```diff
@@ -0,0 +1,38 @@
+//! Wrapper around the `EventSource` API
+//!
+//! This API is provided in the following flavors:
+//! - [Futures API][futures]
+
+pub mod futures;
+
+use std::fmt;
+
+/// The state of the EventSource.
+///
+/// See [`EventSource.readyState` on MDN](https://developer.mozilla.org/en-US/docs/Web/API/EventSource/readyState)
+/// to learn more.
+#[derive(Copy, Clone, Debug)]
+pub enum State {
+    /// The connection has not yet been established.
+    Connecting,
+    /// The EventSource connection is established and communication is possible.
+    Open,
+    /// The connection has been closed or could not be opened.
+    Closed,
+}
+
+/// Error returned by the EventSource
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
```

Comment:
```suggestion
#[derive(Clone, Debug, Eq, PartialEq)]
```
`Copy` means we can't add a error message returned from the `EventSource` in the future

Reply:
This `lib.rs` file was causing this to occur I think:

```rust
#![deny(
    missing_docs,
    missing_debug_implementations,
    missing_copy_implementations
)]
```

Commit 6d6cf4f69e614be46073759182e531f88d6ca0de addreses this by permitting it for the error type.